src/main/java/org/folio/auth/authtokenmodule/tokens/Token.java

```diff
@@ -82,15 +82,41 @@ public String getTenant() {
   public static Future<Token> validate(TokenValidationContext context) {
     Token token;
     try {
-      token = parse(context.getTokenToValidate(), context.getTokenCreator());
+      token = validateTokenContent(context.getTokenToValidate(), context.getTokenCreator());
     } catch (TokenValidationException e) {
       return Future.failedFuture(e);
-    } catch (Exception e) {
-      return Future.failedFuture(new TokenValidationException("Unexpected token parse exception", e, 500));
     }
 
     // Call the validateContext implementation of the underlying token type (AccessToken,
     // RefreshToken, etc.). See those classes for the validation logic specific to each type.
+    return validateTokenType(token, context);
+  }
+
+  /**
+   * Validates if token payload is correct and parsed successfully.
+   * @param tokenToValidate the token content payload to validate
+   * @param tokenCreator the token creator
+   * @return token if token content parsed successfully
+   * @throws TokenValidationException if token content could not be parsed
+   */
+  public static Token validateTokenContent(String tokenToValidate, TokenCreator tokenCreator) throws TokenValidationException {
+    try {
+      return parse(tokenToValidate, tokenCreator);
+    } catch (TokenValidationException e) {
+      throw e;
```

Comment:
I don't feel like this method is necessary. I like the previous code better (probably because I wrote it...) because you don't have to rethrow the same exception. The idea was that `parse` would consolidate all exceptions into the one type of `TokenValidationException` and if that type didn't get caught you would at least have a log message that indicated it was something not expected.

Reply:
Good catch, thank you.
I have deleted this method, now things have become simpler.